### ISSUE TITLE
Resteasy fails with NumberFormatException if file is larger than 2047 MB

### ISSUE THREAD

**Issue by coperator** [quarkusio/quarkus#17455]
## Describe the bug
In my service I receive an InputStream in my REST endpoint and use this inputstream directly to store the data in MinIO.
Everything is working well except if the request contains a "Content Length" header and the file is larger than 2047 MB. In this case a NumberFormatException happens because in `VertxInputStream.java` the header is parsed as int" in method `public int readBytesAvailable()` line, 297.

Stacktrace:
```
java.lang.NumberFormatException: For input string: "3080543800"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:652)
	at java.base/java.lang.Integer.parseInt(Integer.java:770)
	at io.quarkus.vertx.http.runtime.VertxInputStream$VertxBlockingInput.readBytesAvailable(VertxInputStream.java:297)
	at io.quarkus.vertx.http.runtime.VertxInputStream.available(VertxInputStream.java:137)
	at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:359)
	at io.minio.S3Base.getAvailableSize(S3Base.java:1225)
	at io.minio.S3Base.putObject(S3Base.java:1134)
	at io.minio.MinioClient.putObject(MinioClient.java:1666)

```

If no Content Length header is given everythings works fine for large files as well, since the readBytesAvailable method simply returns 0 in this case.

### Output of `java -version`
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.7+10, mixed mode)

### Quarkus version or git rev
1.11.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)

**Comment by coperator** [comment]
Thank you very much for fixing this quickly, @geoand! Would it be possible to backport this fix to 1.11. or 1.13. ?

**Comment by geoand** [comment]
Yeah, that's a good point
